app/services/post_alerter.rb

```diff
@@ -46,7 +46,16 @@ def self.push_notification(user, payload)
     end
 
     if user.push_subscriptions.exists?
-      Jobs.enqueue(:send_push_notification, user_id: user.id, payload: payload)
+      if user.seen_since?(SiteSetting.push_notification_time_window_mins.minutes.ago)
+        Jobs.enqueue_in(
```

Comment:
Does it make sense for this notification to be sent with a delay if the user is already active?

Reply:
We want to be double sure the user is actually active hence usage of the wait window.